lib/query.js

```diff
@@ -1127,9 +1123,7 @@ Query.prototype.findOne = function (conditions, fields, options, callback) {
 
   if (options) {
     this.setOptions(options);
-  }
-
-  if (fields) {
+  } else if (fields) {
```

Comment:
This isn't necessarily correct, both fields and options can be defined. You're right about the other two ifs being redundant though. If you can get rid of this change I can merge this PR